playbooks/juju/pre.yaml

```diff
@@ -36,21 +36,25 @@
       shell: >
         curl https://api.github.com/repos/{{ func_pr_repo.stdout }}/pulls/{{ func_pr_num.stdout }} | \
         awk '/"head":/ {getline; gsub(/"|,/, ""); print $2}'
-      register: func_pr_label
+      register: func_pr_label_cmd
+    - name: func-test pr
+      when: func_pr_label_cmd.rc == 0
+      set_fact:
+        func_pr_label: "{{ func_pr_label_cmd.stdout }}"
 
     - name: Github PR login
-      when: func_pr_label.rc == 0
+      when: func_pr_label is defined
       shell: >
-        echo {{ func_pr_label.stdout }} | cut -f1 -d:
+        echo {{ func_pr_label }} | cut -f1 -d:
       register: github_pr_login
     - name: Github PR branch
-      when: func_pr_label.rc == 0
+      when: func_pr_label is defined
       shell: >
-        echo {{ func_pr_label.stdout }} | cut -f2 -d: | sed -e 's/\//\\\//g'
+        echo {{ func_pr_label }} | cut -f2 -d: | sed -e 's/\//\\\//g'
       register: github_pr_branch
 
     - name: 'handle func-test-pr'
-      when: github_pr_login.rc
+      when: github_pr_login.rc == 0
       shell: >
         GITHUB_PR_REPO="{{ github_pr_repo.stdout }}"
```

Comment:
I think the issue is that I don't see where `github_pr_repo` is being defined and the particular func-test-pr we're looking at is coming from a fork (so a different repo). What do you think?

Reply:
Ah I see, line 19, but maybe you want to do the same trick there?

Reply:
We're only reading the stdout from `github_pr_repo` once, where before we were actually using the `func-pr-label` twice. I don't know that setting it two different times will really buy us much here